src/crimson/os/seastore/random_block_manager/nvme_block_device.h

```diff
@@ -358,9 +356,9 @@ class NVMeBlockDevice : public RBMDevice {
 private:
   // identify_controller/namespace are used to get SSD internal information such
   // as supported features, NPWG and NPWA
-  nvme_command_ertr::future<nvme_identify_controller_data_t> 
+  nvme_command_ertr::future<std::optional<nvme_identify_controller_data_t>> 
```

Comment:
Taking identify_controller / namespace a step further - How about not returning `input_output_error` when unable to get the relevant info?
We could simply use the optional to support both identified data (or not) instead of returning the error.
This could make the users easier to handle no value, pseudo:
```
  auto id_ctr_data = co_await identify_controller(device);
  if (id_ctr_data) {
    // ..logic
  }
```
Without the need for _any_ error handlers.